apertium/apertium_tagger.cc

```diff
@@ -54,768 +29,10 @@
-
-  FILE_Tagger_.train(*ms, TheFunctionTypeOptionArgument);
-  delete ms;
-  close_untagged_files(
-    NULL, UntaggedFn,
-    NULL, UntaggedCorpus);
-
-  Serialised_FILE_Tagger =
-      try_open_file("SERIALISED_TAGGER", ProbFn, "wb");
-  FILE_Tagger_.serialise(Serialised_FILE_Tagger);
-  try_close_file("SERIALISED_TAGGER", ProbFn, Serialised_FILE_Tagger);
-}
-
-void apertium_tagger::s_FILE_Tagger(FILE_Tagger &FILE_Tagger_) {
-  LtLocale::tryToSetLocale();
-
-  if (TheFunctionTypeOptionArgument == 0) {
-    expect_file_arguments(nonoptarg, 5, 7);
-  } else {
-    expect_file_arguments(nonoptarg, 6);
-  }
-  char *DicFn, *CrpFn, *TsxFn, *ProbFn, *TaggedFn, *UntaggedFn;
-  bool do_unsup = nonoptarg == 6;
-
-  get_file_arguments(
-      do_unsup,
-      &DicFn, &CrpFn, &TaggedFn, &UntaggedFn,
-      &TsxFn, &ProbFn);
-  init_FILE_Tagger(FILE_Tagger_, TsxFn);
-
-  FILE *Dictionary, *UntaggedCorpus;
-  MorphoStream* ms = setup_untagged_morpho_stream(
-    FILE_Tagger_,
-    DicFn, UntaggedFn,
-    &Dictionary, &UntaggedCorpus);
-  FILE *TaggedCorpus = try_open_file("TAGGED_CORPUS", TaggedFn, "r");
-  FileMorphoStream tms(TaggedCorpus, true, &FILE_Tagger_.get_tagger_data());
-
-  FILE_Tagger_.init_probabilities_from_tagged_text_(tms, *ms);
-  try_close_file("TAGGED_CORPUS", TaggedFn, TaggedCorpus);
-  delete ms;
-  close_untagged_files(
-    DicFn, UntaggedFn,
-    Dictionary, UntaggedCorpus);
-
-  if (do_unsup) {
-    FILE *Corpus = try_open_file_utf8("CORPUS", CrpFn, "r");
-    FILE_Tagger_.train(Corpus, TheFunctionTypeOptionArgument);
-    try_close_file("CORPUS", CrpFn, Corpus);
- }
-
-  FILE *Serialised_FILE_Tagger =
-      try_open_file("SERIALISED_TAGGER", ProbFn, "wb");
-  FILE_Tagger_.serialise(Serialised_FILE_Tagger);
-  try_close_file("SERIALISED_TAGGER", ProbFn, Serialised_FILE_Tagger);
-}
-
-void apertium_tagger::t_FILE_Tagger(FILE_Tagger &FILE_Tagger_) {
-  LtLocale::tryToSetLocale();
-
-  expect_file_arguments(nonoptarg, 4);
-
-  char *DicFn, *TsxFn, *ProbFn, *UntaggedFn;
-  UntaggedFn = NULL;
-
-  get_file_arguments(
-      false,
-      &DicFn, NULL, NULL, &UntaggedFn,
-      &TsxFn, &ProbFn);
-  init_FILE_Tagger(FILE_Tagger_, TsxFn);
-
-  FILE *Dictionary, *UntaggedCorpus;
-  MorphoStream* ms = setup_untagged_morpho_stream(
-    FILE_Tagger_,
-    DicFn, UntaggedFn,
-    &Dictionary, &UntaggedCorpus);
-
-  FILE_Tagger_.init_and_train(*ms, TheFunctionTypeOptionArgument);
-  delete ms;
-  close_untagged_files(
-    DicFn, UntaggedFn,
-    Dictionary, UntaggedCorpus);
-
-  FILE *Serialised_FILE_Tagger =
-      try_open_file("SERIALISED_TAGGER", ProbFn, "wb");
-  FILE_Tagger_.serialise(Serialised_FILE_Tagger);
-  try_close_file("SERIALISED_TAGGER", ProbFn, Serialised_FILE_Tagger);
-
-}
-}
-
 int main(int argc, char **argv) {
   try {
-    apertium_tagger(argc, argv);
-  } catch (const err_Exception &err_Exception_) {
+    Apertium::apertium_tagger(argc, argv);
+  } catch (const Apertium::err_Exception &err_Exception_) {
```

Comment:
Could this be reverted by moving https://github.com/apertium/apertium/pull/52/files?file-filters%5B%5D=.cc&file-filters%5B%5D=.h&file-filters%5B%5D=.i&file-filters%5B%5D=.in#diff-d30673bddfe5dab2e86c97f0ca3b7351R25 to this file?

Reply:
including `err_exception.h`, and removing `Aperetium::`, is causing error `error: ISO C++ forbids declaration of ‘err_Exception’ with no type [-fpermissive]`

Reply:
Huh... okay. Don't care much.